consumers/writers/timescale/consumer.go

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package timescale
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq" // required for DB access
+	"github.com/mainflux/mainflux/consumers"
+	"github.com/mainflux/mainflux/pkg/errors"
+	mfjson "github.com/mainflux/mainflux/pkg/transformers/json"
+	"github.com/mainflux/mainflux/pkg/transformers/senml"
+)
+
+const (
+	errInvalid        = "invalid_text_representation"
+	errUndefinedTable = "undefined_table"
+)
+
+var (
+	errInvalidMessage = errors.New("invalid message representation")
+	errSaveMessage    = errors.New("failed to save message to timescale database")
+	errTransRollback  = errors.New("failed to rollback transaction")
+	errNoTable        = errors.New("relation does not exist")
+)
+
+var _ consumers.Consumer = (*timescaleRepo)(nil)
+
+type timescaleRepo struct {
+	db *sqlx.DB
+}
+
+// New returns new TimescaleSQL writer.
+func New(db *sqlx.DB) consumers.Consumer {
+	return &timescaleRepo{db: db}
+}
+
+func (pr timescaleRepo) Consume(message interface{}) (err error) {
```

Comment:
This naming (`pr`) is probably copy-paste error from Postgres. You can even use simple `repo`.

Reply:
I'd use `tr` to be consistent with other service

Reply:
Done